plugins/techdocs-backend/src/service/router.ts

```diff
@@ -226,6 +242,23 @@ export async function createRouter(
     );
   });
 
+  // Ensures that the related entity exists and the current user has permission to view it.
+  router.use('/static/docs/:namespace/:kind/:name', async (req, _res, next) => {
```

Comment:
Seems like the order of the middlewares here is quite important.  Might be worth a few test cases to prevent auth regressions on the static objects (e.g. object is cached but user is not auth'd).

Reply:
Pushed a couple tests in 17b9542094dc0a10ff04f2cfc62a8d7475a5ad58